addons/godot_rl_agents/sync.gd

```diff
@@ -263,8 +263,8 @@ func _demo_record_process():
 	current_demo_trajectory[0].append(obs_dict.obs)
 
 	# Get the action applied for the current obs from the agent
-	agent_demo_record.set_action()
 	var acts = agent_demo_record.get_action()
+	agent_demo_record.set_action(acts)
```

Comment:
the actual change is here

Reply:
Ok, after skimming the documentation I think I understand what you did there. The difference between your implementation and mine is that:
* you switch between user input and actual action input at the AIController level
* I switch between the same things at the _player (or robot) level

The reason I did that was such that I never pass a `null` value to the AIController script. But I see the benefits of encapsulating the code as you did. 

Since there's a lot to change and will confuse some of the students, I'll just update the code on my side and close the PR once you see this message.

Reply:
Both approaches can be validly implemented, it's just that one was chosen when implementing the feature and the tutorials for the codebase, and they should be functionally equivalent.